cobalt/browser/application.cc

```diff
@@ -619,19 +619,83 @@ void AddCrashLogApplicationState(base::ApplicationState state) {
              << "required version, so not sending application state.";
 }
 
+void RecordLoaderAppTimeMetrics(
+    const StarboardExtensionLoaderAppMetricsApi* metrics_extension) {
+  int64_t elf_load_duration_us =
+      metrics_extension->GetElfLoadDurationMicroseconds();
+  int64_t elf_decompression_duration_us =
+      metrics_extension->GetElfDecompressionDurationMicroseconds();
+
+  if (elf_load_duration_us < 0) {
+    return;
+  }
+  if (elf_decompression_duration_us < 0) {
+    return;
+  }
+  if (elf_decompression_duration_us > elf_load_duration_us) {
+    // The decompression duration is considered to be contained within the ELF
+    // load duration, so this inequality is unexpected.
+    return;
+  }
+
+  int64_t unexplained_duration_us =
+      elf_load_duration_us - elf_decompression_duration_us;
+
+  base::UmaHistogramTimes(
+      "Cobalt.LoaderApp.ElfLoadDuration",
+      base::TimeDelta::FromMicroseconds(elf_load_duration_us));
+  base::UmaHistogramTimes(
+      "Cobalt.LoaderApp.ElfDecompressionDuration",
+      base::TimeDelta::FromMicroseconds(elf_decompression_duration_us));
+
+  // "Unexplained" just means means that we haven't yet attempted to break this
+  // chunk of time into its components. It's included here to give context for
+  // the other durations, as recommended by
+  // https://chromium.googlesource.com/chromium/src/+/lkgr/docs/speed/diagnostic_metrics.md#summation-diagnostics.
+  base::UmaHistogramTimes(
```

Comment:
Do you have unit tests for these new histogram additions?

Reply:
No. So far I've just tested everything by running Cobalt locally.

I think the best way to add tests would be to extract this logic into a separate class/module and then stub out the dependency on `SbSystemGetExtension()` so that tests can configure the `LoaderAppMetrics` extension as needed. I'd probably prefer to do that in a quick follow-up PR if you're ok with that, as it'll involve some refactoring of existing code.

Reply:
PR here: https://github.com/youtube/cobalt/pull/2825.